libs/blog/shared/util-seo/src/lib/services/seo.service.ts

```diff
@@ -0,0 +1,193 @@
+import { inject, Injectable } from '@angular/core';
+import { Meta, MetaDefinition, Title } from '@angular/platform-browser';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { filter, map, switchMap } from 'rxjs';
+
+import { SeoMetaData } from '@angular-love/contracts/articles';
+
+import { SEO_CONFIG } from '../tokens';
+
+import { SEO_META_KEYS, SeoMetaKeys } from './seo-meta-keys';
+import { SEO_TITLE_KEYS, SeoTitleKeys } from './seo-title-keys';
+
+@Injectable()
+export class SeoService {
+  private readonly _router = inject(Router);
+  private readonly _activatedRoute = inject(ActivatedRoute);
+  private readonly _title = inject(Title);
+  private readonly _meta = inject(Meta);
+  private readonly _seoConfig = inject(SEO_CONFIG);
+
+  constructor() {
+    this.subscribe();
+  }
+
+  setMeta(seoData: SeoMetaData | undefined): void {
+    if (!seoData) {
```

Comment:
I'd consider using `switch`

Reply:
We have to check ALL props and set them. `switch-case` would only set one of the props.